core-java-modules/core-java-io-5/src/test/java/com/baeldung/charstreams/CharStreamsUnitTest.java

```diff
@@ -0,0 +1,116 @@
+    private static final DailyTodo TODO_1 = new DailyTodo("Sleep", LocalDateTime.now(), false);
+    private static final DailyTodo TODO_2 = new DailyTodo("Read", LocalDateTime.now(), false);
+    public static String TODO_LIST_PW_FILE = "src/test/resources/pw_todos.txt";
+    public static String TODO_LIST_FW_FILE = "src/test/resources/fw_todos.txt";
+    
+
+    @BeforeEach
+    public void createFiles() throws IOException {
+        Files.createDirectories(Paths.get(TODO_LIST_PW_FILE).getParent());
+        Files.createDirectories(Paths.get(TODO_LIST_FW_FILE).getParent());
+    }
+
+    @Test
+    public void whenUsingTextData_thenPrintWriterWritesToFile() throws IOException {
+        try {
+            pwFile = CharStreams.writingStringToFilePrintWriter(TODO_LIST_PW_FILE , TODO.toString());
+            assertThat(new File(TODO_LIST_PW_FILE)).hasContent(TODO.toString());
+        }catch (FileNotFoundException fnfe) {
+            fnfe.printStackTrace();
+        }
+    }
+
+    @Test
+    public void whenUsingTextData_thenFileWriterWritesToFile() throws IOException {
+        try {
+            CharStreams.writingStringToFileFileWriter(TODO_LIST_FW_FILE, TODO.toString());
+            assertThat(new File(TODO_LIST_FW_FILE)).hasContent(TODO.toString());
+        } catch (FileNotFoundException fnfe) {
+            fnfe.printStackTrace();
+        }
+        
+    }
+
+    @Test
+    public void whenUsingFormattedData_thenPrintWriterWritesToFile() throws IOException {
+        try {
+            CharStreams.writingNonStringToFilePrintWriter(TODO_LIST_PW_FILE, TODO_1);
+            assertThat(new File(TODO_LIST_PW_FILE)).hasContent(TODO_1.toString());
+        } catch (FileNotFoundException fnfe) {
+            fnfe.printStackTrace();
+        }
+    }
+
+    @Test
+    public void whenUsingFormattedData_thenFileWriterWritesToFile() throws IOException {
+        try {
+            CharStreams.writingNonStringToFileFileWriter(TODO_LIST_FW_FILE, TODO_1);
+            assertThat(new File(TODO_LIST_FW_FILE)).hasContent(TODO_1.toString());
+        } catch (FileNotFoundException fnfe) {
+            fnfe.printStackTrace();
+        }
+    }
+
+    @Test
+    public void whenWritingToExixtingFile_thenPrintWriterAppendsToFile() throws IOException {
+        try {
+            CharStreams.appendingToFilePrintWriter(TODO_LIST_PW_FILE, TODO_2);
+            assertThat(new File(TODO_LIST_PW_FILE)).hasContent(TODO_2.toString());
+        } catch (FileNotFoundException fnfe) {
+            fnfe.printStackTrace();
+        }
+    }
+
+    @Test
+    public void whenWritingToExixtingFile_thenFileWriterAppendsToFile() throws IOException {
+        try {
+            CharStreams.appendingToFileFileWriter(TODO_LIST_FW_FILE, TODO_2);
+            assertThat(new File(TODO_LIST_FW_FILE)).hasContent(TODO_2.toString());
+        } catch (FileNotFoundException fnfe) {
+            fnfe.printStackTrace();
+        }
+    }

```

Comment:
For these two append tests, to truly test append you need to first begin the test with an existing file that has existing content. Then, after the append, we should assert that both the previous content and the appended content is present.

For example:
```
        String existingText = "existing text";
        Files.write(Paths.get(TODO_LIST_FW_FILE), existingText.getBytes(StandardCharsets.UTF_8), StandardOpenOption.CREATE_NEW);

        try (FileWriter fileWriter = new FileWriter(new File(TODO_LIST_FW_FILE), StandardCharsets.UTF_8, true)) {
            fileWriter.append(TODO.toString());
        }

        assertThat(new File(TODO_LIST_FW_FILE)).hasContent(existingText + TODO.toString());
```